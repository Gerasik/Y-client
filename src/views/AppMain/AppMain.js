export default class AppMain {
  constructor(searchEvent) {
    this.searchEvent = searchEvent;
    this.name = 'header-search';
  }

  render() {
    const header = document.createElement('header');
    const searchForm = document.createElement('form');
    const search = document.createElement('input');
    const submit = document.createElement('input');

    search.setAttribute('type', 'text');
    search.setAttribute('placeholder', 'Search');
    submit.setAttribute('type', 'submit');
    submit.setAttribute('value', 'Search');

    header.classList.add(this.name);
    search.classList.add(`${this.name}__search`);

    document.body.appendChild(header);
    header.appendChild(searchForm);
    searchForm.appendChild(submit);
    searchForm.appendChild(search);

    searchForm.addEventListener('submit', this.searchEvent, false);
  }
}
