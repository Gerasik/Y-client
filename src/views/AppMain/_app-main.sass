.header-search
  display: flex
  justify-content: center
  padding: 40px

input
  height: 100%
  font-size: 20px
  padding: 5px
  box-sizing: border-box

input[type='submit']
  color: white
  font-weight: bold
  border: none
  border-radius: 10px 0 0 10px
  background: #dc7a15

input[type='text']
  border: none
  border-radius: 0 10px 10px 0
