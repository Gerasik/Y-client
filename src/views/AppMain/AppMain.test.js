import AppMain from './AppMain';

describe('AppMain.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppMain.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      searchEvent: function a() {
        return 1;
      },
    };
    AppMain.prototype.render.call(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
