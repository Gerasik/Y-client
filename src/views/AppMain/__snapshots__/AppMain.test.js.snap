// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMain.prototype.render Should be render correctly 1`] = `"<header class=\\"undefined\\"><form><input type=\\"submit\\" value=\\"Search\\"><input type=\\"text\\" placeholder=\\"Search\\" class=\\"undefined__search\\"></form></header>"`;
